server.js

```diff
@@ -1,35 +1,131 @@
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+const Book = mongoose.model("Book", {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
+
+
+// RESET_DB=true npm run dev for MongoCompass
+if(process.env.RESET_DB) {
+  const resetDataBase = async () => {
+    await Book.deleteMany();
+    booksData.forEach(singleBook => {
+      const newBook = new Book(singleBook);
+      newBook.save();
+    })
+  }
+  resetDataBase();
+}
+
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+//Routes
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const navigation = {
+    guide: "These are the routes for this book API!",
+    Endpoints: [
+      {
+        "/bookData": "Display all books",
+        "/bookData/authors/:authors": "Search for specific author,  like /authors/Douglas Adams", 
+        "/bookData/title/:title": "Search for a title"
+      },
+    ],
+  };
+  res.send(navigation);
 });
 
-// Start the server
+app.get('/bookData', (req, res) => {
+  res.status(200).json({
+    data: booksData,
+    success: true,
+  })
+  })
+
+ app.get('/bookData/average_rating', async (req, res) => {
+  const { average_rating } = req.params;
+  const bookRating = booksData.sort((a, b) => b.average_rating - a.average_rating)
+    res.json(bookRating.slice(0, [-1])) 
+})
+
+app.get("/bookData/title/:title", async (req, res) => {
+  const { title } = req.params
+
+  try {
+    const bookTitle = await Book.findOne({
+      title : new RegExp(title)
+    })
+
+    if (bookTitle) {
+      res.status(200).json({
+        success: true,
+        results: bookTitle,
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        status_code: 404,
+        message: `Sorry, we don't carry a book by this title: ${title} (* Hint * I'm sensitive!)`
+      })
+    }
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      status_code: 400,
+      message: "Bad request",
+    })
+  }
+})
+
+
+app.get("/bookData/authors/:authors", async (req, res) => {
+  const { authors } = req.params
+
+  try {
+    const authorByName = await Book.findOne({
+      authors : new RegExp(authors)
+    })
+
+    if (authorByName) {
+      res.status(200).json({
+        success: true,
+        results: authorByName,
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        status_code: 404,
+        message: `Sorry, we could not find any book by ${authors}`
```

Comment:
I like that you include the authors name that was searched for in the error message, this way I could easily see when error was caused by spelling incorrectly!